fdbclient/MultiVersionTransaction.actor.cpp

```diff
@@ -1891,13 +1891,25 @@ void MultiVersionApi::loadEnvironmentVariableNetworkOptions() {
 		return;
 	}
 
+	FDBOptionParamType curParamType;
+	int64_t intParamVal;
 	for (auto option : FDBNetworkOptions::optionInfo) {
 		if (!option.second.hidden) {
 			std::string valueStr;
 			try {
 				if (platform::getEnvironmentVar(("FDB_NETWORK_OPTION_" + option.second.name).c_str(), valueStr)) {
+					curParamType = option.second.paramType;
 					for (auto value : parseOptionValues(valueStr)) {
-						Standalone<StringRef> currentValue = StringRef(value);
+						Standalone<StringRef> currentValue;
+						if (curParamType == Int) {
+							char* end;
+							intParamVal = strtoull(value.c_str(),&end,10);
```

Comment:
`strtoull` will parse an unsigned integer, so we'll need to use the signed version of this (`strtoll`). Additionally, the error handling for this case is a little bit complicated and it seems we don't quite follow it everywhere, but it also involves checking `errno`. We could change to do that, or possibly use `stoll` which throws exceptions instead.